### PR TITLE
Add cell attachment support

### PR DESCRIPTION
**Pull request by bgparkerdev**
Adding support for jupyter [cell attachments](https://nbformat.readthedocs.io/en/latest/format_description.html#cell-attachments) to the markdown renderer.

Approach is to use the AST plugin provided by [react markdown](https://github.com/rexxars/react-markdown). 

Since attachments are not contained in the source an additional property attachments had to be added. This is a type of the structure `[attachmentName: string]: string`. Where attachmentName is the key(name) of the attachment and the string is expected to be a url for the image. This is used to look up images and replace the jupyter `attachment:key` with the url assigned to attachmentName(key).

An additional utility method is provided that can take the jupyter nb attachment cell format and convert it to `[attachmentName: string]: string` structure. I'm not sure if this is useful or should be removed.

### CODE REVIEW

**Comment by bgparkerdev** [comment]
@rgbkrk for visibility

**Comment by rgbkrk** [comment]
We should also test that this component works when

* `cellAttachments` is not set (not implemented by an app, or data is missing from the notebook)
* `cellAttachments` is not set even though a `inline image` appears in the markdown
* `cellAttachments` is empty even though a `inline image` appears

File: __tests__/attachment.spec.tsx [hunk 1]
```diff
@@ -19,42 +17,56 @@ function buildSource(attachment?: string): string {
...
 
-  test("Attachments should be added to source if there is matching reference", () => {
+  test("Source is unaffected when it contains jupyter attachment reference and cell attachments is not provided", () => {
+    const wrapped = mount(
+      <MarkdownRender source={buildSource(inlineAttachment)}></MarkdownRender>
+    );
+    const imgProps = wrapped.find("img").props();
+    expect(imgProps).toHaveProperty(
+      "src",
+      "attachment:spot-the-difference-2a.jpg"
```
Comment (rgbkrk) [inline __tests__/attachment.spec.tsx:50]: I'm curious what this is going to be like for users. When their image isn't available, it's going to have a 404 for this URL -- I wonder if we should include an `alt` tag or something on the element that says "cell attachment missing".

Comment (bgparkerdev) [reply __tests__/attachment.spec.tsx:50]: I like the idea of using an alt tag.

Comment (rgbkrk) [reply __tests__/attachment.spec.tsx:50]: As a throwback, I'd love if we had 

image

**Comment by rgbkrk** [comment]
Thank you @bgparkerdev!

**Pull request merged**